src/firefly/java/edu/caltech/ipac/astro/FITSTableReader.java

```diff
@@ -210,53 +209,43 @@ else if (strategy == FULLY_FLATTEN) {
     /**
      * Get a list of star tables from a FITS file.
      * @param fits_filename
+     * @param table_idx
      * @return List of star tables.
      * @throws IOException
      */
-    public static List getStarTableList(String fits_filename) {
 
-        ArrayList<StarTable> tList = new ArrayList<>();
+    public static List getStarTableList(String fits_filename, int table_idx) {
+
+        ArrayList<StarTable> tList =  new ArrayList<>();
 
-        // disable long string for HeaderCard create while collecting tables from StarTableFactory to work around the exception error
-        // sent from nom.tam.fits.
+        // disable long string for HeaderCard creation while collecting table with table_idx from StarTableFactory to work around
+        // the exception error sent from nom.tam.fits.
 
         FitsFactory.useThreadLocalSettings(true);
+        boolean prevLongStringEnabled = FitsFactory.isLongStringsEnabled();
```

Comment:
No need to save/recover that one variable.  You can set `FitsFactory.useThreadLocalSettings(false);` at the end of this method to clear all threadlocal settings.